tests/python/pants_test/backend/jvm/tasks/jvm_compile/test_dep_exports_integration.py

```diff
@@ -0,0 +1,57 @@
+# coding=utf-8
+# Copyright 2017 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import (absolute_import, division, generators, nested_scopes, print_function,
+                        unicode_literals, with_statement)
+
+import os
+import shutil
+
+from pants.base.build_environment import get_buildroot
+from pants_test.pants_run_integration_test import PantsRunIntegrationTest
+
+
+class DepExportsIntegrationTest(PantsRunIntegrationTest):
+
+  SRC_PREFIX = 'testprojects/tests'
+  SRC_TYPES = ['java', 'scala']
+  SRC_PACKAGE = 'org/pantsbuild/testproject/exports'
+
+  @classmethod
+  def hermetic(cls):
+    return True
+
+  def test_compilation(self):
```

Comment:
Might be worth adding a test case demonstrating that these do indeed fail to compile without the `exports` mechanism. Otherwise some change to the test files might cause the compilations in this test to succeed for other reasons, and the test wouldn't be testing what we think it is.